ocaml/seahash/hash.ml

```diff
@@ -0,0 +1,16 @@
+open Ctypes
+open Foreign
+
+let bytes = foreign "bytes" (ocaml_bytes @-> uint64_t @-> returning uint64_t)
+let file = foreign "file" (string @-> returning uint64_t)
+
+let to_hex hash =
+	Printf.sprintf "%016LX" hash;;
+
+let bytes buff = 
+	let size = Unsigned.UInt64.of_int (Bytes.length buff) in
+	Unsigned.UInt64.to_int64(bytes (ocaml_bytes_start buff) size);;
+
+let file name = 
+	close_in (open_in_bin name); (* To emulate the behaviour of the Digest class, also throws an error if the file does not exist *)
```

Comment:
You can use `Sys.file_exists` and `Sys.is_directory` to check the file: https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&ved=2ahUKEwi55qyD6O3fAhULzqQKHdl-Dh8QFjAAegQICRAB&url=https%3A%2F%2Fcaml.inria.fr%2Fpub%2Fdocs%2Fmanual-ocaml%2Flibref%2FSys.html&usg=AOvVaw0ZFy5l_8_hAeiq59UjyXpa